### PR TITLE
s3 replication module

### PR DESCRIPTION
**Pull request by rajkumarpatel2602**
Changes :
1. s3 Replicator and replication manager along with client code for testing.
2. Logging mechanism with rotation
3. Makefile, requirements.txt and flake8 check support
4. Yaml configuration file for replicator and replication manager

TODO :
1. Adding relative path for the log.py
2. Deciding config file placement
-----
[View rendered s3/README.md](https://github.com/rajkumarpatel2602/cortx-multisite/blob/scripts-replicator-scheduler/s3/README.md)

### CODE REVIEW

**Review by kaustubh-d — CHANGES_REQUESTED** [review]

File: s3/replication/manager/src/manager_conf.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+   host: "127.0.0.1"
+   port: 8080
+   ssl: false
+   service: "manager_service"
+   
+logconfig:
+   manager_logger_name: "s3_replication_manager"
+   logger_directory: "/root/dump/"
+   #logger_directory: "/var/log/seagte/s3/replication/manager"          
+   log_file: "manager_logging.out"
```
[the remark this answers is not in the document]
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/manager_conf.yaml:12]: done

File: s3/replication/manager/src/manager_conf.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+manager:
+   host: "127.0.0.1"
+   port: 8080
+   ssl: false
+   service: "manager_service"
+   
+logconfig:
+   manager_logger_name: "s3_replication_manager"
```
[the remark this answers is not in the document]
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/manager_conf.yaml:9]: done

File: s3/replication/manager/src/manager_conf_init.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import sys
+import os
+import shutil
+import logging
+import yaml
+import uuid
+import glob
+import logging.handlers
+
+class config():
```
[the remark this answers is not in the document]
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/manager_conf_init.py:17]: done

File: s3/replication/manager/src/manager_conf_init.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    """ configuratio for manager """
+
+    rep_conf = None
```
[the remark this answers is not in the document]
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/manager_conf_init.py:20]: done

File: s3/replication/manager/src/scheduler.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                LOG.debug('would be response : {}'.format(response))
+            return web.json_response({'Response': response})
+        else:
+            return web.json_response({'Response':'INVALID subscriber'})
+
+    else:
+        return web.json_response({'jobs': list(jobs.keys())})
+
+@routes.get('/jobs/{job_id}')
+async def get_job_attr(request):
```
[the remark this answers is not in the document]
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/scheduler.py:79]: done

File: s3/replication/manager/src/scheduler.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    """Get job attribute
+
+    Handler api to fetch job attributes
+
+    """
+    id = (request.match_info['job_id'])
```
[the remark this answers is not in the document]
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/scheduler.py:85]: done

File: s3/replication/manager/src/scheduler.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+    # Get job's attribute
+    if id in jobs.keys():
+        return web.json_response(jobs[id])
+    else:
+        return web.json_response({'Response':'ERROR : Job is not present!'})
+
+@routes.post('/jobs')
+async def add_jobs(request):
```
[the remark this answers is not in the document]
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/scheduler.py:94]: done

**Comment by rajkumarpatel2602** [comment]
Addressed all the comment along with restructuring changes in this [PR](https://github.com/Seagate/cortx-multisite/pull/5) and Hence closing this PR.

**Pull request closed without merging**